packages/cli/src/utils/init.js

```diff
@@ -28,6 +30,70 @@ const initApp = async (path, createFunc) => {
   await removeDir(tempAppDir);
 };
 
+const testAuth = authType => {
+  // todo: oauth1 trello url?
+  const testUrl =
+    authType === 'digest'
+      ? 'https://httpbin.org/digest-auth/auth/myuser/mypass'
+      : 'https://auth-json-server.zapier-staging.com/me';
+
+  return `
+      const test = (z, bundle) => {
+        return z.request({
+          url: '${testUrl}',
+          ${authType === 'oauth2' ? inspect({}) : ''}
+        })
+      }
+      `.trim();
+};
+
+/**
+ * given an authType (such as `basic` or `oauth2`, return a string of valid JS code that be saved into an `authentication.js` file
+ */
+const createAuthFile = authType => {
+  // TODO: call prettier on this
+
+  const suppliedItem =
+    {
+      custom: 'API Key',
+      oauth2: 'access token'
+    }[authType] || 'username and/or password';
+
+  const output = `
+    ${testAuth(authType)}
+
+    const handleBadResponses = (response, z, bundle) => {
```

Comment:
I don't love that these are stored in strings (there's no syntax highlighting when writing, there's no pretttier), but i'm not sure a better way to do it. We want to build the code in pieces (so being able to drop in conditional strings inline is great), so this is probably the best way to do it. 

The big alternative is to lean on `jscodeshift`, but i'm not sure that'll be much easier to write and iterate with. It's also tougher to read. We could probably reduce code re-use, but it's dense. 

The bottom line is that this'll get prettified before it goes to the user, so it's ok if it's ugly in here as long as its syntactically correct.

Reply:
> I don't love that these are stored in strings

I also feel that. It's a little hard to maintain this way. Can we split these code snippets into template files? It's kind of like what we're already doing with the `scaffold` command, where we keep template files in the [`scaffold`](https://github.com/zapier/zapier-platform/tree/4878770112684c23d175ecfa56076b1dbe8da9a7/packages/cli/scaffold) folder.

I also noticed [previously](https://github.com/zapier/zapier-platform/pull/74) you created an [`examples`](https://github.com/zapier/zapier-platform/tree/4878770112684c23d175ecfa56076b1dbe8da9a7/examples/auth) folder. I forget what was the intention there. Are they going to be used as templates for `init` or are they just "examples"?

Reply:
The examples folder was from a PR distilling every `example-apps` folder into a slim set of files. We unfortunately can't delete `example-apps` because older CLI versions use it on master for the `init` command. 

We can have them in template files for lodash. that'll get us syntax highlighting, but not prettier (since the `<%= %>` tags make it invalid again). If we go this route, we'll do a lot of reading of files and treating them as strings. Might be better, i'll play with it today. 

I think either way, if the result is run through `prettier`, it'll ensure all snippets are syntactically correct and formatted nicely. as long as we have test coverage, we'll at least know if something is super broken.